boilerplate/fullstack/sveltekit/src/superTokensHelpers.ts

```diff
@@ -0,0 +1,279 @@
+    searchParams.forEach((value, key) => {
+        query[key] = value;
+    });
+    return query;
+}
+
+function createPreParsedRequest(request: Request): PreParsedRequest {
+    return new PreParsedRequest({
+        cookies: getCookieFromRequest(request),
+        url: request.url as string,
+        method: request.method as HTTPMethod,
+        query: getQueryFromRequest(request),
+        headers: request.headers,
+        getFormBody: async () => {
+            return await request.formData();
+        },
+        getJSONBody: async () => {
+            return await request.json();
+        },
+    });
+}
+
+async function getSessionDetails(
+    preParsedRequest: PreParsedRequest,
+    options?: VerifySessionOptions,
+    userContext?: Record<string, unknown>
+): Promise<{
+    session: SessionContainer | undefined;
+    hasToken: boolean;
+    hasInvalidClaims: boolean;
+    baseResponse: CollectingResponse;
+    svelteResponse?: Response;
+}> {
+    const baseResponse = new CollectingResponse();
+    // Possible interop issue.
+    const recipe = (SessionRecipe as any).default.instance;
+    const tokenTransferMethod = recipe.config.getTokenTransferMethod({
+        req: preParsedRequest,
+        forCreateNewSession: false,
+        userContext,
+    });
+    const transferMethods = tokenTransferMethod === "any" ? availableTokenTransferMethods : [tokenTransferMethod];
+    const hasToken = transferMethods.some((transferMethod) => {
+        const token = getToken(preParsedRequest, "access", transferMethod);
+        if (!token) {
+            return false;
+        }
+        try {
+            parseJWTWithoutSignatureVerification(token);
+            return true;
+        } catch {
+            return false;
+        }
+    });
+
+    try {
+        const session = await Session.getSession(preParsedRequest, baseResponse, options, userContext);
+        return {
+            session,
+            hasInvalidClaims: false,
+            hasToken,
+            baseResponse,
+        };
+    } catch (err) {
+        if (Session.Error.isErrorFromSuperTokens(err)) {
+            return {
+                hasToken,
+                hasInvalidClaims: err.type === Session.Error.INVALID_CLAIMS,
+                session: undefined,
+                baseResponse,
+                svelteResponse: new Response("Authentication required", {
+                    status: err.type === Session.Error.INVALID_CLAIMS ? 403 : 401,
+                }),
+            };
+        } else {
+            throw err;
+        }
+    }
+}
+
+/**
+ * A helper function to retrieve session details on the server side.
+ *
+ * NOTE: This function does not use the getSession function from the supertokens-node SDK
+ * because getSession can update the access token. These updated tokens would not be
+ * propagated to the client side, as request interceptors do not run on the server side.
+ */
+export async function getSessionForSSR(svelteRequest: Request): Promise<{
```

Comment:
I assume you got all these functions from somewhere.. these functions are now in the node repo itself from the supertokens-node/nextjs import path. I know this is a sveltekit app, but for now, you can try and import these functions from this path and see if it works. If it does, we can expose those functions via some other way from the sdk to make it more generic.

This includes functions like handleAuthAPIRequest. Bascially all these functions should not be needed for the user to implement.

Reply:
Maybe this would be better to to in a new PR, since we have some version of this in at least 5 integrations. It might make more sense to fix them via a dedicated PR.

Reply:
ok